src/RedisMessaging.ReliableDelivery/Subscribe/Message.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace RedisMessaging.ReliableDelivery.Subscribe
+{
+    public struct Message
+    {
+        public long Id { get; }
+        public string Content { get; }
+
+        public Message(long id, string content)
+        {
+            Id = id;
+            Content = content;
+        }
+
+        public static Message Undefined { get; } = default(Message);
+
+        public override string ToString()
+        {
+            return $"Message:Id={Id}";
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            return obj is Message other && Equals(other);
+        }
+
+        public bool Equals(Message other)
+        {
+            return Id == other.Id;

```

Comment:
Shouldn't equal use both fields
```suggestion
            return Id == other.Id && string.Equals(Content, other.Content);
```